tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,105 +1,81 @@
 
 describe(__filename, () => {
-  const render = (props) => {
+  const getProps = ({ i18n = fakeI18n(), ...props } = {}) => {
+    return {
+      i18n,
+      id: 'showMoreCard',
+      store: dispatchClientMetadata().store,
+      ...props,
+    };
+  };
+
+  function render({ children, ...otherProps } = {}) {
+    const props = getProps(otherProps);
     return shallowUntilTarget(
-      <ShowMoreCard i18n={fakeI18n()} {...props} />,
+      <ShowMoreCard {...props}>{children || 'some info'}</ShowMoreCard>,
       ShowMoreCardBase,
     );
-  };
+  }
 
   it('reveals more text when clicking "show more" link', () => {
-    const root = render({ children: 'Hello I am description' });
-
-    // We have to manually set the expanded flag to false because we don't have
-    // a clientHeight in the tests.
-    root.setState({ expanded: false });
+    const { store } = dispatchClientMetadata();
+    const root = render({ store });
 
     expect(root).not.toHaveClassName('ShowMoreCard--expanded');
 
-    // We have to shallow this prop because we pass a component to it.
-    expect(shallow(root.prop('footerLink')).html()).toContain(
-      '<span class="visually-hidden">Expand to</span> Read more',
-    );
-
     // We cannot directly target the footer link, so we simulate the click here.
     root.instance().onClick(createFakeEvent());
-    root.update();
 
-    expect(root).toHaveClassName('ShowMoreCard--expanded');
+    applyUIStateChanges({ root, store });
 
-    expect(root).toHaveProp('footerLink', null);
+    expect(root).toHaveClassName('ShowMoreCard--expanded');
   });
 
-  it('is expanded by default', () => {
-    const root = render({ children: 'Hello I am description' });
+  it('is unexpanded by default', () => {
+    const root = render();
 
-    expect(root).toHaveClassName('ShowMoreCard--expanded');
+    expect(root).not.toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('truncates the contents if they are too long', () => {
-    const root = render({ children: 'Hello I am description' });
+    const root = render();
 
+    // We are simulating the truncate method call on did mount.
     root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
     root.update();
-
     expect(root).not.toHaveClassName('ShowMoreCard--expanded');
   });
 
   it('renders className', () => {
     const className = 'test';
-    const root = render({
-      children: <p>Hi</p>,
-      className,
-    });
-
+    const root = render({ className });
     expect(root).toHaveClassName(className);
   });
 
   it('renders children', () => {
     const root = render({ children: 'Hello I am description' });
-
-    expect(root.childAt(0)).toHaveText('Hello I am description');
+    const contents = root.find('.ShowMoreCard-contents');
+    expect(contents).toHaveText('Hello I am description');
   });
 
-  it('executes truncateToMaxHeight when it recieves props', () => {
-    const root = mount(<ShowMoreCardBase i18n={fakeI18n()} />);
```

Comment:
I explained why I removed this above. let me know if you disagree / feel this IS needed..